src/sentry/templatetags/sentry_react.py

```diff
@@ -95,10 +95,15 @@ def get_react_config(context):
         messages = []
         is_superuser = False
 
+    enabled_features = []
+
     if user:
         user = extract_lazy_object(user)
+    if not user.is_anonymous():
+        enabled_features = [project.organization.slug + ':' + project.slug + ':similarity-view'
+            for project in Project.objects.filter(organization=User.objects.get(id=user.id).get_orgs()) if
+            features.has('projects:similarity-view', project=project)]
```

Comment:
Ok, so the query itself isn't O(n). Not sure what influence I was under.

But this query is going to be scoped incorrectly. The problem is it's not being scoped by team membership. You're saying, "any organizations this user is a part of, get all of the projects." which is invalid without checking Team membership. You may be a part of an org, but not a part of all of the teams.

This is where `Project.objects.get_for_user` comes into play, but takes `team` as an argument.

So without Team being present here, this is a bad check.